src/SDL_hints.c

```diff
@@ -34,123 +31,153 @@ typedef struct SDL_HintWatch
 
 typedef struct SDL_Hint
 {
-    char *name;
     char *value;
     SDL_HintPriority priority;
     SDL_HintWatch *callbacks;
-    struct SDL_Hint *next;
 } SDL_Hint;
 
-static SDL_Hint *SDL_hints;
+static SDL_PropertiesID hint_props = 0;
```

Comment:
```suggestion
static SDL_PropertiesID SDL_hint_props = 0;
```

Reply:
Fixed in a5e9faf2937d746b6438d1c3849b46d75b66f650.